### ISSUE TITLE
Support setting the default transaction isolation level for major relational databases

### ISSUE THREAD

**Issue by jn-deque** [typeorm/typeorm#7545]
### The Problem

While TypeORM does support setting the [transaction isolation level through the API ](https://github.com/typeorm/typeorm/blob/master/docs/transactions.md)for specific transactions, it would be very convenient to set a default transaction isolation level for *all* connections/transactions when first calling `createConnection()`.

### The Solution

For relational databases that support the standard isolation levels the API could support something like the following:
```
    const conn = await createConnection({
        "type": "mysql",
        "host": "localhost",
        "port": 3306,
        "username": "my_user",
        "password": "some_pass",
        "database": "my_db",
        "synchronize": false,
        "migrationsRun": false,
        "isolationLevel": "READ UNCOMMITTED"
     });
```

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [x] `mysql`
- [ ] `nativescript`
- [x] `oracle`
- [x] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [x] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [x] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [x] No, I don't have the time and I wouldn't even know how to start.

**Comment by michaelmontero** [comment]
any update on this?

**Comment by 3ximus** [comment]
According to the documentation it's possible (at least for sqlserver)
https://orkhan.gitbook.io/typeorm/docs/data-source-options#mssql-data-source-options
Using `options.connectionIsolationLevel`

**Comment by martin-scholz-emnify** [comment]
i don't think it is available for mysql its a shame please add it.

**Comment by AmrAnwar** [comment]
It would be nice to have it, thanks in advance

**Comment by vahidvdn** [comment]
That would be great to have. Any update on this?

**Comment by alumni** [comment]
Does this even make sense? If you change the default isolation level, you're automatically in a transaction. How would we commit/rollback that transaction? Also due to the fact that many drivers pick a random connection from the pool, you wouldn't have any isolation at all, even with this setting.

Maybe would it only make sense for multi-query operations, like when using the Repository API to load relations via separate queries?

CC: @pkuczynski @Cprakhar

**Comment by pkuczynski** [comment]
> Does this even make sense? If you change the default isolation level, you're automatically in a transaction. How would we commit/rollback that transaction? Also due to the fact that many drivers pick a random connection from the pool, you wouldn't have any isolation at all, even with this setting.

Yeah, I think it does make sense. Have a look how I implemented this in #12269. It only uses this settings when you kick in a transaction, unless you specify another isolation level for this specific one. It does not mean that it creates transactions for any operations.
